### PR TITLE
OD-253 [Feature] Filter value type User profile data

### PR DESCRIPTION
**Pull request by AndrRyaz**
@sofiiakvasnevska 
## Issue
OD-253 https://weboo.atlassian.net/jira/software/projects/OD/boards/9?selectedIssue=OD-253
## Description
Added logic for filter value type, user profile data to in order to follow the user profile data fields
## Screenshots/screencasts
https://screencast-o-matic.com/watch/cYlb2Z8vgg
## Backward compatibility
This change is fully backward compatible.
## Reviewers
@upplabs-alex-levchenko
## Notes
- link to the previous approder PR - https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/464
- Added fixes to  avoid bugs that happened last time with merge this PR

The PR's should be merged in the sequence OD-252,254,258

### CODE REVIEW

**Comment by tonytlwu** [comment]
@AndrRyaz If there's a commit missing, you need to get that working and showing in the PR. Create a new PR if you have to.

Though, in https://github.com/AndrRyaz/fliplet-widget-dynamic-lists/commit/1bf1d27f220ef37d1a3bafb5dd9fbd75f93ad90d there's also the problem of having too many related selectors.

```js
        $newPanel.find('.panel-title-text .value, #value-dash, #filter-value-type').addClass('hidden');
        $newPanel.find('.panel-title-text .value, #value-dash, #filter-value').addClass('hidden');
```

`.panel-title-text .value, #value-dash` is repeated.

**Comment by AndrRyaz** [comment]
@tonytlwu in PR https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/490/files#diff-3e00189c2b38ee2b81964659c3a01eceef437f47aeefe02a20fa6f8d876a938fL1940 this lines will be removed.
Should i make this fixed in current PR.

File: js/interface.js [hunk 1]
```diff
@@ -1929,7 +1928,6 @@ var DynamicLists = (function() {
       $filterAccordionContainer.append($newPanel);
 
       if (data.logic === 'empty' || data.logic === 'notempty') {
-        $newPanel.find('.panel-title-text .value, #value-dash, #filter-value-type').addClass('hidden');
```
Comment (tonytlwu) [inline js/interface.js:1932]: The `#filter-value-type` UI is not relevant when data.log is `empty` or `notempty`. This means we can't remove this line of code.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/interface.js (hunk 1 above)
Comment (AndrRyaz) [reply js/interface.js:1932]: @tonytlwu, fixed

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: LFDs with filters configured in Studo
Verify: All existing filters work as configured

Squash and merge: Adds support for filtering with user profile data

**Pull request merged**